src/function/mtk.jl

```diff
@@ -25,7 +27,7 @@ function instantiate_function(f, x, ad::AutoModelingToolkit, p, num_cons=0)
 
     if f.hv === nothing
         hv = function (H, θ, v, args...)
-            res = ArrayInterfaceCore.zeromatrix(θ)
+            res = ad.obj_sparse ? sparse(findnz(hess_prototype)[1], findnz(hess_prototype)[2], zeros(nnz(hess_prototype))) : ArrayInterfaceCore.zeromatrix(θ)
```

Comment:
I don't understand this. It should just be the cache copy of `hess_prototype`?

Reply:
actually, `hv` should be very different if you're generating the Hessian code. You should just `build_function` on the result of `H*v`.

Reply:
That gives a `Bool` eltype so that's why I had to create these here

Reply:
>actually, hv should be very different if you're generating the Hessian code. You should just build_function on the result of H*v.

I see, makes sense, I'll try that out